src/main/java/org/spongepowered/common/command/SpongeCommandManager.java

```diff
@@ -118,36 +129,64 @@ public SpongeCommandManager(Logger logger, Disambiguator disambiguator) {
-                    }
-                }
-
-                aliasesWithPrefix.add(alias);
+            List<String> aliasesWithPrefix = this.calculatePrefixesWithNamespace(container, aliases);
 
-                // Alias commands with unqualified ID and qualified ID
-                String unqualifiedId = container.getUnqualifiedId();
-                aliasesWithPrefix.add(unqualifiedId + ':' + alias);
+            Optional<CommandMapping> mapping = this.dispatcher.register(callable, aliasesWithPrefix, callback);
 
-                if (!container.getId().equals(unqualifiedId)) {
-                    aliasesWithPrefix.add(container.getId() + ':' + alias);
-                }
+            if (mapping.isPresent()) {
+                this.owners.put(container, mapping.get());
             }
 
-            Optional<CommandMapping> mapping = this.dispatcher.register(callable, aliasesWithPrefix, callback);
+            return mapping;
+        }
+    }
+
+    @Override
+    public void install(Object plugin, AbstractCommandModule module) {
+        checkNotNull(plugin, "plugin");
+        checkNotNull(module, "module");
+        checkState(module.canBeInstalledOrConfigured(true), "The provided command module has already been installed.");
+
+        Optional<PluginContainer> containerOptional = Sponge.getGame().getPluginManager().fromInstance(plugin);
+        if (!containerOptional.isPresent()) {
+            throw new IllegalArgumentException(NO_ASSOCIATED_CONTAINER);
+        }
+
+        module.configure(containerOptional.get(), this);
+    }
+
+    @Override
+    public void register(AbstractCommandModule module, Object object) {
+        checkNotNull(module, "module");
+        checkState(module.canBeInstalledOrConfigured(false), "The provided command module has been fully configured and "
+                + "cannot be used to register commands.");
+        checkNotNull(object, "object");
+
+        synchronized (this.lock) {
+            this.annotationCommandFactory.register(module, object);
+        }
+    }
+
+    @Override
+    public Optional<CommandMapping> register(AbstractCommandModule module, CommandCallable callable, List<String> aliases,
+            Function<List<String>, List<String>> callback) {
+        checkNotNull(module, "module");
+        checkState(module.canBeInstalledOrConfigured(false), "The provided command module has been fully configured and "
+                + "cannot be used to register commands.");
+
+        PluginContainer container = module.getPlugin();
+
+        synchronized (this.lock) {
+            List<String> aliasesWithPrefix = this.calculatePrefixesWithNamespace(container, aliases);
+
+            Optional<CommandMapping> mapping = this.dispatcher.register(callable, aliasesWithPrefix, callback,
+                    (cc, al) -> new ImmutableModuleCommandMapping(module, callable, aliases.get(0), aliases.subList(1, aliases.size())));
```

Comment:
In the API half you had something similar extracted out to a field, not sure if you wanted it done here or not. 

https://github.com/SpongePowered/SpongeAPI/pull/1201/files#diff-89adf4090b06b371ecc8b77db21c4abfR89